### PR TITLE
Newanims

### PR DESCRIPTION
**Pull request by Scrndles**

### CODE REVIEW

File: Classes/CZ805M.uc [hunk 1]
```diff
@@ -2,11 +2,323 @@ class CZ805M extends MP7MMedicGun;
...
+    local int MyAmmos;
+    
+    //don't do any of this stuff on servers
+    if ( (Level.NetMode == NM_DedicatedServer) || (AIController(Instigator.Controller) != None) )
+		return;
+    
+    MyAmmos = MagAmmoRemaining;
+    if (MyAmmos > 9)
+    {
+        MagFloat = (MyAmmos-10)/(MagCapacity-10);
```
Comment (poosh) [inline Classes/CZ805M.uc]: I don't remember if MagCapacity is int or float, but if it is int, then int/int = int, hence MagFloat will be either 0 or 1.

Comment (poosh) [reply Classes/CZ805M.uc]:
```suggestion
        MagFloat = float(MyAmmos-10)/(MagCapacity-10);
```

File: Classes/CZ805M.uc [hunk 2]
```diff
@@ -2,11 +2,323 @@ class CZ805M extends MP7MMedicGun;
...
+    }
+    if (MyAmmos < 1)
```
Comment (poosh) [inline Classes/CZ805M.uc:51]: So many ifs here. How about creating an array of struct with ammo and bone name fields. Then you could simply iterate through the array.

File: Classes/CZ805MFire.uc [hunk 3]
```diff
@@ -5,22 +5,23 @@ defaultproperties
...
+     maxVerticalRecoilAngle=150
      maxHorizontalRecoilAngle=75
-     ShellEjectClass=Class'ROEffects.KFShellEjectSCAR'
+     //ShellEjectClass=Class'ROEffects.KFShellEjectSCAR'
+     ShellEjectClass=class'ScrnWeaponPack.CZ805MShellEject' //changed to m4
      ShellEjectBoneName="Shell_eject"
      bAccuracyBonusForSemiAuto=True
      DamageType=Class'ScrnWeaponPack.DamTypeCZ805M'
-     DamageMax=35
+     DamageMax=41
```
Comment (poosh) [inline Classes/CZ805MFire.uc:17]: Why did you buff CZ805 damage? The initial idea was to have its low base damage and high headshot multiplier. With all your visual fixes to the weapon, I think it is even more important to stay like that. 
If you want to make it stronger, consider raising headshot multiplier instead of base damage.

File: Classes/CZ805MPickup.uc [hunk 4]
```diff
@@ -3,7 +3,7 @@ class CZ805MPickup extends KFWeaponPickup;
 defaultproperties
 {
      Weight=6.000000
-     cost=3250
+     cost=3000
```
Comment (poosh) [inline Classes/CZ805MPickup.uc:6]: Why this was needed? I set it to 3250 to make the price difference from other guns.

File: Classes/DamTypeCZ805M.uc [hunk 5]
```diff
@@ -17,7 +17,7 @@ static function AwardDamage(KFSteamStatsAndAchievements KFStatsAndAchievements,
 
 defaultproperties
 {
-     HeadShotDamageMult=1.5
+     HeadShotDamageMult=1.2
```
Comment (poosh) [inline Classes/DamTypeCZ805M.uc:20]: Once again, why transfer headshot to bae damage?

File: Classes/HRLProj.uc [hunk 6]
```diff
@@ -3,6 +3,7 @@ class HRLProj extends ScrnLAWProj;
 defaultproperties
 {
     ExplosionClass=Class'ScrnWeaponPack.HRLExplosion'
+    ImpactDamage=275
```
Comment (poosh) [inline Classes/HRLProj.uc:6]: Good change. It makes sense that rockets do more impact damage than grenades, and LAW has a stronger impact than HRL.

File: Classes/Protecta.uc [hunk 7]
```diff
@@ -14,41 +14,6 @@ exec function SwitchModes()
     DoToggle();
 }
 
-simulated function bool CanZoomNow()
-{
-    // aiming while firing makes zoom bugged. Prevent zooming while firing.
-    return Instigator != none && Instigator.Controller.bFire == 0;
```
Comment (poosh) [inline Classes/Protecta.uc:20]: Are you sure this isn't needed anymore? This is the third attempt to get fix zooming bug :)

File: Classes/SVDSPickup.uc [hunk 8]
```diff
@@ -5,7 +5,7 @@ class SVDSPickup extends KFWeaponPickup;
 
 defaultproperties
 {
-     Weight=7.000000
+     Weight=8.000000
```
Comment (poosh) [inline Classes/SVDSPickup.uc]: Why? SVDS was intended to be used with other 7kg weapons, such as Hunting Rifle.

File: Classes/ScopedWeapon.uc [hunk 9]
```diff
@@ -224,11 +232,29 @@ simulated function UpdateScopeMode()
                 ScriptedScopeCombiner.AlphaOperation = AO_Use_Mask;
                 ScriptedScopeCombiner.Material2 = ScopeScriptedTexture;
             }
-
+    
+			if( IllumTex != none && ScriptedScopeStatic == none )
```
Comment (poosh) [inline Classes/ScopedWeapon.uc]: Messed up tabulation

File: Classes/ScrnRPGAltFire.uc [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+    //do nothing
+}
+
+event ModeHoldFire()
+{
+}
+
+simulated function bool AllowFire()
+{
+    //if( Level.TimeSeconds - LastClickTime>FireRate )
```
Comment (poosh) [inline Classes/ScrnRPGAltFire.uc]: clean up the code please

**Review by poosh — CHANGES_REQUESTED** [review]
Wow! I'm impressed. These changes will bring ScrN Weapon Pack to a new level.
However, there are a few issues I'd like you to address before the merge.

File: Classes/HRLProj.uc (hunk 6 above)
Comment (Scrndles) [reply Classes/HRLProj.uc:6]: this change is also quite significant, previously demo perked headshots would only do 639, just barely enough to stun a scrake. After this change HRL will be able to stun scrakes and LAW will be able to stun jasons, this makes a big difference for demos engaging charging scrakes within arming range by allowing a stun with a blunt headshot which was only possible with RPG before due to 500 base impact damage (which is now lowered to 400)
The blunt damage change was to make the other launchers consistent with the RPG, but if this makes demos vs charging scrakes too easy perhaps all impact damage will have to be tweaked

File: Classes/Protecta.uc (hunk 7 above)
Comment (Scrndles) [reply Classes/Protecta.uc:20]: the zooming bug was caused by bad idle positions and bad recoil animations, both are fixed in the new animation package so this isn't needed anymore

File: Classes/CZ805MFire.uc (hunk 3 above)
Comment (Scrndles) [reply Classes/CZ805MFire.uc:17]: I think the idea of a medic rifle compared to a medic SMG is that it has higher base damage and possibly headshot damage, the tradeoff is the rifles weigh more and also do not reload as quickly as SMGs. Setting the damage value to 41 makes the CZ805M have higher damage than all the other SMGs and makes it consistent with the M4 M203M, this will also allow for commando and combat medic perked CZ805Ms to 3shot HOE crawlers offperk and 2shot them on perk which gives it an edge compared to SMGs when it comes to ammo efficiency against them. The 1.5x headshot multiplier out of nowhere felt strangely high and made bodyshots feel comparatively too useless. I set it to 1.2x since the M4 has 1.3x and a few other commando weapons have headshot multipliers above 1.1x, the CZ805's can be raised to 1.3x without issues but any higher would be unfair to other commando weapons

File: Classes/CZ805MPickup.uc (hunk 4 above)
Comment (Scrndles) [reply Classes/CZ805MPickup.uc:6]: I think the CZ805M should not get reload speed bonus with medic perks, medics currently reload it faster than commando while also having a 100% magazine size bonus compared to commando's 35%. In the event that the reload speed bonus is removed, the price of the weapon should be reduced to compensate, but most of the CZ805M's purpose is having high medic dart charge and low dart cost so the price was not reduced that much.

File: Classes/SVDSPickup.uc (hunk 8 above)
Comment (Scrndles) [reply Classes/SVDSPickup.uc]: Given the SVDS's ability to stun jasons, high scope zoom, fast reload speed and quick refire time I thought this balance change was necessary as this weapon is a complete upgrade compared to the hunting rifle, and can compete decently with the M14. If you could afford the SVDS there was little reason to buy the hunting rifle as the only difference is bodyshot damage and scope zoom, I thought this change would let hunting rifle remain a viable choice.
It was also to make the SVDS weight closer to the SVD, since I think it's strange there was a weight difference of 4 between the weapons, which will be 2 if SVDS weight is increased and SVD weight is decreased

File: Classes/CZ805M.uc (hunk 2 above)
Comment (Scrndles) [reply Classes/CZ805M.uc:51]: It would certaintly be a nicer solution but there wasn't anything to copypaste and I found it easier to just write all the lines instead.

File: Classes/CZ805M.uc (hunk 1 above)
Comment (Scrndles) [reply Classes/CZ805M.uc]: During testing the magfloats variable worked as intended, but there's no harm in adding it anyway for good practice

File: Classes/CZ805M.uc (hunk 2 above)
Comment (poosh) [reply Classes/CZ805M.uc:51]: OK, I'll refactor this myself, so you'll have a source for future copypasting :smile:

File: Classes/CZ805MFire.uc (hunk 3 above)
Comment (poosh) [reply Classes/CZ805MFire.uc:17]: Your arguments sound reasonable. OK, let's apply your changes.

File: Classes/CZ805MPickup.uc (hunk 4 above)
Comment (poosh) [reply Classes/CZ805MPickup.uc:6]: I'll check the medic reload speed bonus after the merge and remove or change it if it will look OP.

File: Classes/Protecta.uc (hunk 7 above)
Comment (poosh) [reply Classes/Protecta.uc:20]: OK, we'll see :smile:

File: Classes/SVDSPickup.uc (hunk 8 above)
Comment (poosh) [reply Classes/SVDSPickup.uc]: I agree that 4 block weight difference between SVDS and SVD looks weird, but it made for balance reasons, mainly to prevent using HC+SVD because that loadout would outshine XBow. Also, keep in mind that, theoretically, weapon weight includes ammo weight, and SVD allows carrying more ammo than SVDS and has a twice bigger magazine.

SVDS is better than HR (Hunting Rifle) in all aspects but the price and Fleshpound killing ability. Thanks to lower damage reduction, HR kills 6p HoE FP in 4 shots while SVDS requires 6. If there is no Demo in the team, then one of the Sharpies picking HR and staying with it till the Boss wave is a viable choice.

And the last but not least, SVDS is not meant to be a replacement of HR but an **addition** to it. With both guns weighting 7 blocks, you can carry both of them while wearing a Combat Armor. It is a way to move from Sharpshooter into a pure Sniper, in games where all small zeds are taken out by teammates, and you have to worry about Husks and above. I'm using HR+SVDS combo myself on maps with long hallways, such as Bedlam, Wicked City or Train.

Long story short, let keep the weight to 7 blocks.

**Review by poosh — APPROVED** [review]
I accept your CZ805M balance proposal. Revert SVDS weight change and it'll be good to go.

File: Classes/SVDSPickup.uc (hunk 8 above)
Comment (Scrndles) [reply Classes/SVDSPickup.uc]: oh, you're right. I completely forgot about the fact that hunting rifle's damage type doesn't suffer damage reduction against large zeds. I think the SVD should get more ammo than it has currently though, it's only 70 right now vs 60 against SVDS, and the SVD's weight prevents you from carrying pretty much any other sharp rifle so I think it should have more base ammo to compensate, maybe that can be considered in the next patch

Comment (poosh) [reply Classes/SVDSPickup.uc]: I agree. It could be raised to 80, maybe even more.
